File: frontend/src/types/shoppingList.ts
/**
 * 買い物リスト型定義
 */

export interface ShoppingListItem {
  itemId: string;
  userId: string;
  name: string;
  quantity: number;
  unit: string;
  isChecked: boolean;
  isCheckedAt: string | null;
  addedAt: string;
  sourceRecipeId: string | null;
}

export interface AddShoppingListItemRequest {
  name: string;
  quantity: number;
  unit: string;
  sourceRecipeId?: string;
}

export interface UpdateShoppingListItemRequest {
  isChecked: boolean;
}

export interface ShoppingListState {
  items: ShoppingListItem[];
  loading: boolean;
  error: string | null;
}

File: backend-python/app/models/shopping_list.py
from dataclasses import dataclass, field
from datetime import datetime
from decimal import Decimal


@dataclass
class ShoppingListItem:
    item_id: str
    user_id: str
    name: str
    quantity: Decimal
    unit: str
    is_checked: bool = False
    is_checked_at: datetime | None = None
    added_at: datetime = field(default_factory=datetime.now)
    source_recipe_id: str = ""
    normalized_key: str = ""
